alarm/utils.py

```diff
@@ -0,0 +1,122 @@
+from alarm.models import Coin, Candle
+import websocket
+import json
+import ssl
+
+
+def connect_binance_socket(currencies, intervals):
+    websocket.enableTrace(False)
+    socket_urls = [f'wss://stream.binance.com:9443/ws/{currency}@kline_{interval}' for currency, interval in
+                   zip(currencies, intervals)]
+
+    sockets = []
+    for socket_url in socket_urls:
+        socket = websocket.create_connection(socket_url, sslopt={'cert_reqs': ssl.CERT_NONE})
+        socket.on_message = feedback_message_from_websocket
+        socket.on_close = close_streaming_message_from_websocket
+        sockets.append(socket)
+
+    return sockets
+
+
+def process_binance_messages(sockets):
+    try:
+        while True:
+            for socket in sockets:
+                message = socket.recv()
+                handle_binance_message(message)
+    except KeyboardInterrupt:
+        close_binance_sockets(sockets)
+    except (ValueError, KeyError) as err:
+        print(err)
+
+
+def parse_binance_message(message):
+    """
+    Parse a Binance WebSocket message and return the relevant data.
+
+    :param message: The WebSocket message to parse.
+    :return: A tuple containing the current candle high price, current candle low price, and coin abbreviation.
+    """
+    json_message = json.loads(message)
+    current_candle = json_message['k']
+    current_candle_high_price = float(current_candle['h'])
+    current_candle_low_price = float(current_candle['l'])
+    coin_symbol = current_candle['s']
+    coin_abbreviation = coin_symbol.lower()
+
+    return current_candle_high_price, current_candle_low_price, coin_abbreviation
+
+
+def handle_binance_message(message):
+    """
+    Handle a Binance WebSocket message.
+
+    :param message: The WebSocket message to handle.
+    """
+    try:
+        # Parse the message to extract relevant data
+        current_candle_high_price, current_candle_low_price, coin_abbreviation = parse_binance_message(message)
+
+        # Get the coin from the database
```

Comment:
61-75 is an atomic operation, updating coin last candle. Essentially that should be a method of a Coin model. See how comment essentialy converts to a method name. Generally, comment should explain why we wrote that code, not what code does. If comment explains what code does - it is a bad comment and we possibly need to refactor naming or structure.

Reply:
++